code/modules/reagents/chemistry/reagents/other_reagents.dm

```diff
@@ -1801,6 +1801,38 @@
 	taste_description = "blueyalty" //also intentional
 	carpet_type = /turf/open/floor/carpet/royalblue
 
+//the ultimate fertilizer
+/datum/reagent/genesis
+	name = "Genesis Serum"
+	description = "A mysterious substance capable of spontaneously gestating plant life when given a surface to adhere to."
+	color = "#328242"
+	taste_description = "primordial essence"
+	reagent_state = LIQUID
+	var/ground_type = /turf/open/floor/grass
+
+/datum/reagent/genesis/expose_turf(turf/T, reac_volume)
+	if(istype(T, /turf/open/floor/grass))//prevents spamming effect via. smoke or such
```

Comment:
```suggestion
	if(istype(T, ground_type))//prevents spamming effect via. smoke or such
```
I'd either replace this or remove ground_type entirely and just put grass in the other reference to it.

Reply:
yeagh. Comitting now